### ISSUE TITLE
key annotations incorrectly initialized (or colorkey incorrectly drawn)

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1243]
**Reported by ansley b manke on 28 Sep 2012 19:38 UTC**
When a color plot is overlaid on an existing plot and has all-missing data, the key annotations are incorrect. They've got values from the previous, underlying plot.

yes? use etopo60
yes? shade/lev=v/pal=grayscale/nokey rose
yes? keymark,1
yes? let xpts =  {1,2,3}
yes? let ypts =  {1,2,3}
yes? let var =  {,,}
yes? GO polymark polygon/over/key/title="SST"/lev=(1,2,1)/pal=ten_by_levels  xpts,ypts,var,square,0.5
 
There are no polygons plotted, but there is a colorbar, annotated at the top and bottom with the range from the ETOPO datset.  The annotations should not be put on the key.

This does not happen with a SHADE plot having all missing data, overlaid on a first valid plot. In that case, no colorbar is made.

But really, should the behavior be that for a POLYGON plot, if there is no valid data, we get no colorbar?

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/1971

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 22 Oct 2012 22:35 UTC**
I propose that we add the same logic to SHADE and POLYGON plots that has been in place for many years for FILL plots.

For FILL plots, there is this control of color keys, in xeq_contour.F:

           IF ( no_data ) do_key = .FALSE.  

It's been in the code base since 1999. It overrides any setting made with FILL/KEY and it means that for any plot drawn as a color fill, whether an overlay or not, if there's no valid data being FILLed, there will not be a color key.

Then all kinds of overlays of no-valid-data will not have these fairly meaningless color keys.

Examples - all of the plots made by this script have all-missing-data overlays and so have no color keys.


```
keymark 1
use etopo60

! FILL plots: all-missing overlay

let all_miss = if rose ge 20000 then rose
shade/lev=v/pal=grayscale/nokey rose
fill/over/key all_miss; pause

shade/lev=v/pal=grayscale/nokey rose
fill/over/key/lev=(0,30000,10000) all_miss; pause


! SHADE plots: all-missing overlay

shade/lev=v/pal=grayscale/nokey rose
shade/over/key all_miss; pause

shade/lev=v/pal=grayscale/nokey rose
shade/over/key/lev=(0,30000,10000) all_miss; pause

! Color polygon overlay

let xpts = {1,2,3}
let ypts = {1,2,3}
let var = {}
shade/lev=v/pal=grayscale/nokey rose
GO polymark polygon/over/key/title="var"/lev=v/pal=ten_by_levels xpts,ypts,var,square,0.5
pause

shade/lev=v/pal=grayscale/nokey rose
GO polymark polygon/over/key/title="var" xpts,ypts,var,square,0.5
```

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 22 Oct 2012 22:43 UTC**
That is, in the example script given in the comment above,  "all of the plots made by this script have all-missing-data overlays and so SHOULD have no color keys".

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 10 Apr 2013 00:03 UTC**
Fixed in v6.84 release

**Comment by karlmsmith** [comment]
**Attachment from ansley.b.manke on 28 Sep 2012 19:39 UTC**
The polygon overlay has incorrectly annotated color key
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1971_GIT_1243/bug1971.gif
